internal/metrics/querynode.go

```diff
@@ -0,0 +1,273 @@
+			Help:      "Number of readers per collection in QueryNode.",
+		}, []string{
+			"num_readers",
+		})
+
+	QueryNodeSearchCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "search_count",
+			Help:      "Search_count request statistic in QueryNode.",
+		}, []string{
+			"status",
+		})
+
+	QueryNodeRetrieveCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "retrieve_count",
+			Help:      "Retrieve request statistic in QueryNode.",
+		}, []string{
+			"status",
+		})
+
+	QueryNodeQueryDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "query_duration",
+			Help:      "The query durations in QueryNode, query means search or retrieve.",
+			Buckets:   queryNodeDurationBuckets,
+		}, []string{
+			"query_type",
+		})
+
+	QueryNodeQueryDurationInUnsolvedBuffer = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "query_duration_in_unsolved",
+			Help:      "The query durations in unsolved buffer in QueryNode, query means search or retrieve.",
+			Buckets:   queryNodeDurationBuckets,
+		}, []string{
+			"query_type",
+		})
+
+	QueryNodeQueryDurationOnSealedSegments = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "query_duration_on_sealed",
+			Help:      "The query durations on sealed segments, query means search or retrieve.",
+			Buckets:   queryNodeDurationBuckets,
+		}, []string{
+			"query_type",
+		})
+
+	QueryNodeQueryDurationOnGrowingSegments = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "query_duration_on_growing",
+			Help:      "The query durations on growing segments, query means search or retrieve.",
+			Buckets:   queryNodeDurationBuckets,
+		}, []string{
+			"query_type",
+		})
+
+	QueryNodeQueryDurationInCore = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "query_duration_in_core",
+			Help:      "The query durations in core, query means search or retrieve.",
+			Buckets:   queryNodeDurationBuckets,
+		}, []string{
+			"query_type",
+		})
+
+	QueryNodeTranslateHitsDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.QueryNodeRole,
+			Name:      "translate_hits_durations",
+			Help:      "The search durations in translate hits.",
+			Buckets:   queryNodeDurationBuckets,
+		}, []string{
+			"translate_hits_duration",
+		})
+
+	QueryNodeReduceDuration = prometheus.NewHistogramVec(
```

Comment:
rename to QueryNodeReduceLantency 
use label to distinguish search and query

Reply:
fixed